autocorpus/bioc_supplementary.py

```diff
@@ -0,0 +1,311 @@
+"""This module provides functionality for converting text extracted from various file types into a BioC format."""
+
+import datetime
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+from pandas import DataFrame
+
+from .ac_bioc import (
+    BioCCollection,
+    BioCDocument,
+    BioCJSON,
+    BioCPassage,
+)
+from .ac_bioc.bioctable import (
+    BioCTableCell,
+    BioCTableCollection,
+    BioCTableDocument,
+    BioCTableJSON,
+    BioCTablePassage,
+)
+
+
+def extract_table_from_pdf_text(text: str) -> tuple[str, list[DataFrame]]:
+    """Extracts tables from PDF text and returns the remaining text and tables.
+
+    Args:
+        text (str): The input text extracted from a PDF file.
+
+    Returns:
+        tuple: A tuple containing the remaining text (str) and a list of tables (list of DataFrames).
+    """
+    # Split the text into lines
+    lines = [x for x in text.splitlines() if x]
+    temp_text_output = lines
+
+    # store extracted tables
+    tables = []
+    # Identify where the table starts and ends by looking for lines containing pipes
+    table_lines = []
+    # keep unmodified lines used in tables. These must be removed from the original text
+    lines_to_remove = []
+    inside_table = False
+    for line in lines:
+        if "|" in line:
+            inside_table = True
+            table_lines.append(line)
+            lines_to_remove.append(line)
+        elif (
+            inside_table
+        ):  # End of table if there's a blank line after lines with pipes
+            inside_table = False
+            tables.append(table_lines)
+            table_lines = []
+            continue
+
+    for line in lines_to_remove:
+        temp_text_output.remove(line)
```

Comment:
Rather than storing a list of the lines you *don't* want, I'd just append the ones you *do* want to a list.